### PR TITLE
Updated remote crates.bzl template to use 'maybe' and fixed Buildifier defects.

### PR DESCRIPTION
**Pull request by UebelAndre**
This is a followup on comments made in #199 

The `crates.bzl` file generated in the event that no crates are defined looks like the following
```python
"""
@generated
cargo-raze crate workspace functions

DO NOT EDIT! Replaced on runs of cargo-raze
"""

load("@bazel_tools//tools/build_defs/repo:git.bzl", "new_git_repository")  # buildifier: disable=load
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")  # buildifier: disable=load
load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")  # buildifier: disable=load

def raze_fetch_remote_crates():
    """No crates were detected in the source Cargo.toml. This is a no-op"""
    pass
```

### CODE REVIEW

**Comment by UebelAndre** [comment]
@damienmg if you have a quick moment. I'd love a review. (This is small). I tried very hard to keep indentation of the template code but was not able to get it to work. Thus, I came up with what I'm presenting to you in this pull-request. The one additional cleanup task I did was to replace the macros that were written in the template file with [maybe](https://docs.bazel.build/versions/master/repo/utils.html#maybe) (which does he same thing).

**Comment by UebelAndre** [comment]
The scope of this PR grew to include fixing the Buildifier defects in the generated Bazel files. I've updated all the templates and added a lint step to CI.

File: impl/src/templates/crate.BUILD.template [hunk 1]
```diff
@@ -4,42 +4,45 @@ cargo-raze crate build file.
...
-  # Prefer access through "{{workspace.workspace_path}}", which limits external
-  # visibility to explicit Cargo.toml dependencies.
-  "//visibility:public",
-])
-
-licenses([
-  "{{crate.license.rating}}", # {{crate.license.name}}
-])
 
+# buildifier: disable=load
```
Comment (damienmg) [inline impl/src/templates/crate.BUILD.template:8]: For my information, what is wrong here that we need this disable?

File: impl/src/templates/partials/remote_crates_patch.template [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
             {% for patch_cmd in crate.raze_settings.patch_cmds -%}
             "{{patch_cmd}}",
             {%- endfor %}
         ],
 {%- endif %}
-{%- if crate.raze_settings.patch_cmds_win %}
+{%- if crate.raze_settings.patch_cmds_win  %}
         patch_cmds_win = [
-            {% for patch_cmd in crate.raze_settings.patch_cmds_win -%}
+            {%  for patch_cmd in crate.raze_settings.patch_cmds_win -%}
```
Comment (damienmg) [inline impl/src/templates/partials/remote_crates_patch.template:24]: Shouldn't that change do nothing? Do you need those spaces? (also in other place). The spacing inside the template bracket is only to respect indentation.

File: impl/src/templates/crate.BUILD.template (hunk 1 above)
Comment (UebelAndre) [reply impl/src/templates/crate.BUILD.template:8]: We can't guarantee that all these macros are used in a generated file. If we don't always disable this warning we'd currently get a wall of the following:
```command
buildifier -lint=warn -r examples
```
```output
examples/vendored/complicated_cargo_library/cargo/vendor/MacTypes-sys-2.1.0/BUILD:10: load: Loaded symbol "rust_binary" is unused. Please remove it.
To disable the warning, add '@unused' in a comment. (https://github.com/bazelbuild/buildtools/blob/master/WARNINGS.md#load)
examples/vendored/complicated_cargo_library/cargo/vendor/MacTypes-sys-2.1.0/BUILD:12: load: Loaded symbol "rust_test" is unused. Please remove it.
To disable the warning, add '@unused' in a comment. (https://github.com/bazelbuild/buildtools/blob/master/WARNINGS.md#load)
examples/vendored/complicated_cargo_library/cargo/vendor/aho-corasick-0.6.10/BUILD:12: load: Loaded symbol "rust_test" is unused. Please remove it.
To disable the warning, add '@unused' in a comment. (https://github.com/bazelbuild/buildtools/blob/master/WARNINGS.md#load)
examples/vendored/complicated_cargo_library/cargo/vendor/arrayvec-0.3.25/BUILD:10: load: Loaded symbol "rust_binary" is unused. Please remove it.
To disable the warning, add '@unused' in a comment. (https://github.com/bazelbuild/buildtools/blob/master/WARNINGS.md#load)
examples/vendored/complicated_cargo_library/cargo/vendor/arrayvec-0.3.25/BUILD:12: load: Loaded symbol "rust_test" is unused. Please remove it.
To disable the warning, add '@unused' in a comment. (https://github.com/bazelbuild/buildtools/blob/master/WARNINGS.md#load)
examples/vendored/complicated_cargo_library/cargo/vendor/arrayvec-0.4.10/BUILD:10: load: Loaded symbol "rust_binary" is unused. Please remove it.
To disable the warning, add '@unused' in a comment. (https://github.com/bazelbuild/buildtools/blob/master/WARNINGS.md#load)
```

File: impl/src/templates/partials/remote_crates_patch.template (hunk 2 above)
Comment (UebelAndre) [reply impl/src/templates/partials/remote_crates_patch.template:24]: I does do nothing. I wasn't sure why you made your suggestion in #199 so I was trying to preempt that here. I'd be happy to remove all these extra spaces if you're fine with it.

Comment (damienmg) [reply impl/src/templates/partials/remote_crates_patch.template:24]: In #199, the indentation was removed to prevent it from surfacing in the result file, I was suggesting to move that indentation inside the tag to compensate and keep the code readable. Here it is not needed.

Comment (UebelAndre) [reply impl/src/templates/partials/remote_crates_patch.template:24]: Ok, I've cleaned up that commit to only include spacing that affects alignment.

Comment (damienmg) [reply impl/src/templates/partials/remote_crates_patch.template:24]: This precise space is still here :)

File: impl/src/templates/remote_crates.bzl.template [hunk 3]
```diff
@@ -4,43 +4,41 @@ cargo-raze crate workspace functions
...
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")  # buildifier: disable=load
 
 def {{workspace.gen_workspace_prefix}}_fetch_remote_crates():
 {%- if crates %}
-    {% for crate in crates %}
-        {%- if crate.source_details.git_data %}
-    _new_git_repository(
+    """This function defines a collection of repos and should be called in a WORKSPACE file"""
+{%- for crate in crates %}
+    {%- if crate.source_details.git_data %}
```
Comment (damienmg) [inline impl/src/templates/remote_crates.bzl.template:16]:
```suggestion
{%-   for crate in crates %}
    {%-   if crate.source_details.git_data %}
```

File: impl/src/templates/remote_crates.bzl.template [hunk 4]
```diff
@@ -4,43 +4,41 @@ cargo-raze crate workspace functions
...
         name = "{{workspace.gen_workspace_prefix}}__{{crate.pkg_name | replace(from="-", to="_")}}__{{crate.pkg_version | slugify | replace(from="-", to="_")}}",
         remote = "{{crate.source_details.git_data.remote}}",
         commit = "{{crate.source_details.git_data.commit}}",
         build_file = Label("{{workspace.workspace_path}}/remote:{{crate.pkg_name}}-{{crate.pkg_version}}.{{workspace.output_buildfile_suffix}}"),
         init_submodules = True,
         {%- include "templates/partials/remote_crates_patch.template" %}
     )
-        {%- else %}
-    _new_http_archive(
+    {%- else %}
```
Comment (damienmg) [inline impl/src/templates/remote_crates.bzl.template:26]:
```suggestion
    {%-   else %}
```

File: impl/src/templates/remote_crates.bzl.template [hunk 5]
```diff
@@ -4,43 +4,41 @@ cargo-raze crate workspace functions
...
-        {% endif -%}
+        {%  endif -%}
         strip_prefix = "{{crate.pkg_name}}-{{crate.pkg_version}}",
         {%- include "templates/partials/remote_crates_patch.template" %}
         build_file = Label("{{workspace.workspace_path}}/remote:{{crate.pkg_name}}-{{crate.pkg_version}}.{{workspace.output_buildfile_suffix}}"),
     )
-        {%- endif %}
-    {% endfor %}
+    {%- endif %}
+{%  endfor %}
```
Comment (damienmg) [inline impl/src/templates/remote_crates.bzl.template:40]:
```suggestion
    {%-   endif %}
{%    endfor %}
```

File: impl/src/templates/crate.BUILD.template [hunk 6]
```diff
@@ -4,41 +4,43 @@ cargo-raze crate build file.
...
+
+licenses([
+    "{{crate.license.rating}}",  # {{crate.license.name}}
+])
+
+# Generated targets
+{%- set crate_name_sanitized = crate.pkg_name | replace(from="-", to="_") %}
 {%- if crate.build_script_target %}
-{%- include "templates/partials/build_script.template" %}
+{%  include "templates/partials/build_script.template" %}
```
Comment (damienmg) [inline impl/src/templates/crate.BUILD.template:30]: Here and below, take the opportunity to show the indentation:
```suggestion
{%    include "templates/partials/build_script.template" %}
```

**Review by damienmg — CHANGES_REQUESTED** [review]
I also wonder, the CI pass here despite having a breaking commit, correct? Or do I read that wrong?

**Comment by UebelAndre** [comment]
> I also wonder, the CI pass here despite having a breaking commit, correct? Or do I read that wrong?

Yeah, this is a problem I'm working on. I likely should have converted this PR to a draft. I believe I've solved the problem though. Buildifier writes it's findings to `stderr` so my subprocess wasn't properly capturing it

File: impl/src/templates/partials/remote_crates_patch.template (hunk 2 above)
Comment (UebelAndre) [reply impl/src/templates/partials/remote_crates_patch.template]: Sorry, I thought I should leave that to align with the `{%- endfor %}` below it. This has been updated 😅

**Comment by UebelAndre** [comment]
@damienmg I applied all your suggestions but I'm not sure why the PR is still saying there's still 1 change requested...

**Comment by UebelAndre** [comment]
Is there a better way to test Travis CI other than having to push changes and wait? I can build much faster locally 😜

**Comment by UebelAndre** [comment]
@damienmg Ok, now I think this is ready for review again 😄

File: impl/src/templates/crate.BUILD.template [hunk 7]
```diff
@@ -4,41 +4,44 @@ cargo-raze crate build file.
...
+{%- set crate_name_sanitized = crate.pkg_name | replace(from="-", to="_") %}
 {%- if crate.build_script_target %}
-{%- include "templates/partials/build_script.template" %}
+{%    include "templates/partials/build_script.template" %}
 {%- endif %}
 {%- for target in crate.targets %}
 {%- set target_name_sanitized = target.name | replace(from="-", to="_") %}
 {%- if target.kind == "lib" %}
-{%- include "templates/partials/rust_library.template" %}
+{%  include "templates/partials/rust_library.template" %}
```
Comment (damienmg) [inline impl/src/templates/crate.BUILD.template:36]: Here and below: also take the opportunity to fix the indentation?

**Review by damienmg — APPROVED** [review]
Just one nit left :)

File: impl/src/templates/crate.BUILD.template (hunk 7 above)
Comment (UebelAndre) [reply impl/src/templates/crate.BUILD.template:36]: Like, just remove the spaces within the brackets? I still don't think I understand what the indentation is supposed to look like 😅

Comment (damienmg) [reply impl/src/templates/crate.BUILD.template:36]: The current logic is hard to follow, Actually there should be space before too since there is a for. So for this line that would be introducing 4 spaces:

```suggestion
{%      include "templates/partials/rust_library.template" %}
```

The idea is to be able to read the template in a way that is logical, for instance I completely missed the fact that all those `if`s  were included into a for loop because of missing indentation.

Comment (UebelAndre) [reply impl/src/templates/crate.BUILD.template]: Like so? da41d6ad671f3c94b330c2dc026df5fdb50bc462

**Review by damienmg — APPROVED** [review]
Thanks that's perfect!
